src/algebra/algebra/spectrum.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2021 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import algebra.algebra.basic
+import tactic.noncomm_ring
+/-!
+# Spectrum of an element in an algebra
+This file develops the basic theory of the spectrum of an element of an algebra.
+This theory will serve as the foundation for spectral theory in Banach algebras.
+
+## Main definitions
+
+* `resolvent a : set R`: the resolvent set of an element `a : A` where
+  `A` is an  `R`-algebra.
+* `spectrum a : set R`: the spectrum of an element `a : A` where
+  `A` is an  `R`-algebra.
+
+## Main statements
+
+* `smul_eq_smul`: units in the scalar ring commute (multiplication) with the spectrum.
+* `left_add_coset_eq`: elements of the scalar ring commute (addition) with the spectrum.
+* `units_mem_mul_iff_mem_swap_mul` and `preimage_units_mul_eq_swap_mul`: the units
+  (of `R`) in `σ (a*b)` coincide with those in `σ (b*a)`.
+
+## Notations
+
+* `σ a` : `spectrum R a` of `a : A`
+
+## TODO
+
+* Prove the *spectral mapping theorem* for the polynomial functional calculus
+-/
+
+universes u v
+
+
+section defs
+
+variables (R : Type u) {A : Type v}
+variables [comm_ring R] [ring A] [algebra R A]
+
+-- definition and basic properties
+
+/-- Given a commutative ring `R` and an `R`-algebra `A`, the *resolvent* of `a : A`
+is the `set R` consisting of those `r : R` for which `r•1 - a` is a unit of the
+algebra `A`.  -/
+def resolvent (a : A) : set R :=
+{ r : R | is_unit (algebra_map R A r - a) }
+
+
+/-- Given a commutative ring `R` and an `R`-algebra `A`, the *spectrum* of `a : A`
+is the `set R` consisting of those `r : R` for which `r•1 - a` is not a unit of the
+algebra `A`.
+
+The spectrum is simply the complement of the resolvent.  -/
+def spectrum (a : A) : set R :=
+(resolvent R a)ᶜ
+
+end defs
+
+variables {R : Type u} {A : Type v}
+variables [comm_ring R] [ring A] [algebra R A]
+
+-- products of scalar units and algebra units
+
+lemma is_unit.smul_iff {G : Type u} [group G] [mul_action G A]
```

Comment:
This lemma does not belong in this file, and should be called `is_unit_smul_iff`. Could you make another PR to move it to a better place, probably in the `group_action` folder?

I wasn't expecting someone else to merge this so quickly!

Reply:
sure, no problem